### PR TITLE
fixed update / added comments / Implemented Create Student

### PR DESCRIPTION
**Pull request by cblanto7**

### CODE REVIEW

**Comment by toasterbrain** [comment]
I notice you changed the html form to use the "post" method rather than the "put" method. Wont this make it so update creates a new student rather than editing the previous student?

**Comment by cblanto7** [comment]
@toasterbrain  no you cannot use PUT in the current setup of our project at all for anything so we will use post instead. See this post for details https://stackoverflow.com/questions/29017593/spring-4-restcontroller-not-receiving-data-from-put-request/29018821#29018821

File: src/main/java/unbanner/StudentController.java [hunk 1]
```diff
@@ -11,6 +11,7 @@
     @Autowired
     StudentRepository repository;
 
+    //list all students
```
Comment (zeonin) [inline src/main/java/unbanner/StudentController.java]: More documentation.

What does 'all' mean? Where do 'all students' come from? What is the listing format?

File: src/main/java/unbanner/StudentController.java [hunk 2]
```diff
@@ -19,21 +20,23 @@ public String StudentsList(Model model) {
 
     @RequestMapping("/student/{id}")
 
+    //show one student
```
Comment (zeonin) [inline src/main/java/unbanner/StudentController.java]: More documentation.

Which one student is showed? What is the format of the lookup string?

File: src/main/java/unbanner/StudentController.java [hunk 3]
```diff
@@ -19,21 +20,23 @@ public String StudentsList(Model model) {
...
     public String student(@PathVariable String id, Model model) {
         model.addAttribute("student", repository.findOne(id));
         return "student";
     }
 
-
+    //delete one student
```
Comment (zeonin) [inline src/main/java/unbanner/StudentController.java]: More documentation.

Same as above, what's the format?

**Comment by cblanto7** [comment]
Create Student now in here #44

File: src/main/java/unbanner/StudentController.java [hunk 4]
```diff
@@ -3,37 +3,93 @@
...
     model.addAttribute("students", repository.findAll());
     return "students";
   }
 
-  @RequestMapping("/student/{id}")
+  /*
+  *  Routing for create_student.html template
+  *  Specifically for HTTP GET requests
+  *  Provides ability for the template to create
+  *  a new student
```
Comment (toasterbrain) [inline src/main/java/unbanner/StudentController.java]: blah

File: src/main/java/unbanner/StudentController.java [hunk 5]
```diff
@@ -3,37 +3,93 @@
...
+  @RequestMapping(value = "/students/new", method = RequestMethod.GET)
+  public String provideStudent(@ModelAttribute("student") Student student) {
+    return "create_student";
+  }
 
+  /*
+  *  Routing for create_student.html template
+  *  Specifically for HTTP POST requests
+  *  Provides ability to save a new
+  *  students to the repository
```
Comment (toasterbrain) [inline src/main/java/unbanner/StudentController.java]: jpi;jk;klj

**Pull request merged**